### ISSUE TITLE
Type can't be selected when submitting a faulty template

### ISSUE THREAD

**Issue by h122015** [MISP/MISP#883]
I've noticed that when defining a template and the form is submitted without defining all required fields, it is not possible to correct the problem since the Type drop-down doesn't show any options. What I did:
1.- Define a new template via "Add Template"
2.- Add a new element to it via the "+" button
3.- Choose "Attribute" from the popup
4.- Fill up "Name" and "Description"
5.- Select "Targetting data" from the Category drop-down
6.- Click "Submit"

Then I get the following error:

The object(s) could not be saved. Please, try again.

And a hint is shown stating what the problem is, namely "Please choose a type". If I click on the "Type" drop-down, it is empty and I have no way of solving the problem.

**Comment by BenjaminJoly** [comment]
I've just tried to reproduce the bug as described in the issue on CentOS 7, php 7.1.11 and MISP 2.4.86.  
Moreover, for me, after the failed attempt to validate, when I do not have any options on the "Type" drop-down, if I just click on submit, I've got an infinite loading. If I click on the "Type" drop-down, I can only choose "Select a type". Once submit, I've got a blank type. 
But by editing the value, everything is back to normal. 
Note : I can also choose no category, to have a blank type and category. No errors shown.
